blog/src/components/CreatePost/CreatePost.jsx

```diff
@@ -0,0 +1,166 @@
+
+  const titleValidation = (e) => {
+    const { value } = e.target;
+    if (value.length < 8) { setErrorTitle('* Required at least 8 characters.'); } else { setErrorTitle(null); }
+  };
+  const imgurlValidation = (e) => {
+    const { value } = e.target;
+    if (value.length < 8) { setErrorImgurl('* Required at least 8 characters.'); }
+    if (isProperImageURL(value) === false) { setErrorImgurl('* Incorrect URL format'); } else { setErrorImgurl(null); }
+  };
+  const contentValidation = (e) => {
+    const { value } = e.target;
+    if (value.length < 100) { setErrorContent('* Required at least 100 characters.'); } else { setErrorContent(null); }
+  };
+  const slugValidation = (e) => {
+    const { value } = e.target;
+    if (value.length < 5) { setErrorSlug('* Required at least 5 characters.'); } else { setErrorSlug(null); }
+  };
+
+  const validateAll = () => {
+    if (!title || title.length < 8) {
+      toast.error('title incorrect: min length 8', { position: toast.POSITION.TOP_RIGHT, duration: 5000 });
+      return false;
+    }
+    const c = isProperImageURL(imgUrl);
+    if (!imgUrl || imgUrl.length < 8 || c === false) {
+      toast.error('Image Url incorrect', { position: toast.POSITION.TOP_RIGHT, duration: 5000 });
+      return false;
+    }
+    if (!content || content.length < 100) {
+      toast.error('content incorrect: min length 100', { position: toast.POSITION.TOP_RIGHT, duration: 5000 });
+      return false;
+    }
+    if (!slug || slug.length < 5) {
+      toast.error('slug incorrect: min length 8', { position: toast.POSITION.TOP_RIGHT, duration: 5000 });
+      return false;
+    }
+    return true;
+  };
+
+  const handlesubmit = (e) => {
+    e.preventDefault();
+    const all = validateAll();
+    if (all === false) {
+      return false;
+    }
+    alert('validations working fine');
+    return true;
+  };
+  const areFieldfilled = () => {
+    if (!title || !imgUrl || !content || !slug) {
+      return false;
+    }
+    return true;
+  };
+
+  return (
+    <section data-testid="createpost">
+      <ToastContainer />
+      <div className="container mb-5">
+        <h4 className='"title title-lg"'>Write a Post</h4>
+        <form className="text-black">
+          <div className="form-group">
+            <label>Title</label>
+            <input
+              type="text"
+              data-testid="inputtitle"
+              className="form-control"
+              placeholder="Title..."
+              value={title}
+              onChange={(e) => (setTitle(e.target.value))}
+              onBlur={titleValidation}
+            />
+            {errorTitle && <small style={{ color: 'red' }}>{errorTitle}</small>}
+          </div>
+
+          {imgUrl.length > 0
+            ? (
+              <div className="customing">
+                <img data-testid="Image" src={imgUrl} alt={title} className="object-fit-cover" />
+              </div>
+            )
+            : null}
```

Comment:
Good job showing the image. However, it is necessary to show an image only if it is a valid image URL also if the image URL is valid but is not found then a proper message should be shown.